### DISCUSSION TITLE
ESP32 v1.19.1 (xQueueGenericSend)- assert failed!

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#9173] [Using MicroPython] [answered]
When upgrading to the v1.19.1 ESP32 Generic image I now get this error:

```
connecting to hotspot...
/home/micropython/esp-idf-v4.2/components/freertos/queue.c:743 (xQueueGenericSend)- assert failed!
```
Happens on ESP32-D0WDQ6-V3 (revision 3) and revision 1 chips.

There is more not working, ie the GPIO.  

Did I miss something important when moving from v1.18 to v1.91.1?

**Comment by dpgeorge** [comment]
> When upgrading to the v1.19.1 ESP32 Generic image I now get this error:

Can you show the Python code that leads to this assert failure?  If you can provide a minimal example of Python code that shows the bug then other people can try to reproduce it and track down the problem.

> There is more not working, ie the GPIO.

As above, can you provide a simple example using `machine.Pin` that shows the problem, what it does on v1.18 and what changed in v1.19.1?

**Comment by davefes** [comment]
I had previously noticed (and forgotten about) this issue on v1.19.1 and also Glenn20's older v1.19 image.  It behaves as expected in his latest v1.19.1 image.
The problem is when you define a static IP ... I find that you need to put in a short delay before it or it crashes.

```
import network
import utime
import machine
import config


sta_if = network.WLAN(network.STA_IF) # create station interface


def connect():
    count = 0


    utime.sleep(1)

    if not sta_if.isconnected():
        print('connecting to hotspot...')
        sta_if.active(True)
        utime.sleep(.5) #   important or it crashes
        sta_if.ifconfig((config.WiFi_device, '255.255.255.0', config.gateway, '8.8.8.8'))

        try:
            sta_if.connect(config.hotspot, config.password)
        except OSError as error:
            try:
                with open('errors.txt', 'a') as outfile:
                    outfile.write(str(error) + '\n')
            except OSError:
                pass

        while (count < 5):
            count += 1

            if (sta_if.isconnected()):
                count = 0
                print (' network config:', sta_if.ifconfig())
                break

            print ('.', end = '')
            utime.sleep(1)


    if (count == 5):
        try:
            with open('errors.txt', 'a') as outfile:
                outfile.write('failed to connect after 5 tries' + '\n')
        except OSError:
            pass

        disconnect() # or you could get errors
        machine.reset() #  start again


    count = 0 #  reset count

    utime.sleep(1)



connect()

```

I'll go back and check the GPIO as I would like to try PULL_HOLD.
Thanks,
Dave

**Reply by dpgeorge** [reply]
> I'll go back and check the GPIO as I would like to try PULL_HOLD.

PULL_HOLD never worked correctly and has been removed from the esp32 port.  There is now a `hold=` keyword argument to the Pin constructor.  See docs http://docs.micropython.org/en/latest/esp32/quickref.html#pins-and-gpio

**Comment by davefes** [comment, marked as answer]
I have just verified that hold=True and hold=False on v1.19.1 work.  Sorry, didn't note the change in keywords.
